---
title: "Bulk LNCap +/- Pou2f3"
author: "Caden McQuillen"
date: '2023-05-26'
output: html_document
---

## Read in feature counts
The count matrix generated by feature counts is read into R to perform exploratory analysis. SRR numbers are replaced with experimental conditions for each sample. Counts matrix is log2 transformed
```{r}
library(DESeq2)
counts_matrix <- read.table("/Users/Caden/Desktop/Pou2f3_Data/LNCap_Pou2f3_treatment/Counts.txt", header = TRUE, row.names = 1)
metadata <- str_split(colnames(counts_matrix), pattern = "\\.")
metadata <- do.call(rbind, metadata)

colnames(metadata) <- c("Genotype", "TP", "Treatment", "Replicate")
metadata<-as.data.frame(metadata)
metadata$POU2F3 <- ifelse(metadata$Genotype == "NRP" | metadata$Genotype == "CNP", "POU2F3+", "POU2F3-")
#metadata<-t(metadata)
#colnames(metadata)<- colnames(counts_matrix)

log_counts_matrix <-log2(counts_matrix)
```




## DEseq object
Counts matrix is used as input for DEseq object. Genes with 0 reads are removed.Size factors were estimated using DEseq estimateSizeFactors().Log2 and Log2 normalized counts matrices were added to DEseq object. Within condition whole gene expression dot plots were generated using log2 normalized counts matrix to check for visual correlation between replicates.
```{r}
library(magrittr)


#DEseq requires intergers 
counts_matrix <- round(counts_matrix, 0)
#Create DEseq object
DESeq.ds <- DESeqDataSetFromMatrix(countData = as.matrix(counts_matrix),
colData = metadata,
design = ~ POU2F3)

#reads per sample
colSums(counts(DESeq.ds))

#remove genes with no reads
keep_genes <- rowSums(counts(DESeq.ds)) > 0
DESeq.ds <- DESeq.ds[ keep_genes, ]
dim(DESeq.ds)


#Size factor
DESeq.ds <- estimateSizeFactors(DESeq.ds) # calculate SFs, add them to object
plot( sizeFactors(DESeq.ds), colSums(counts(DESeq.ds)), # assess them
ylab = "library sizes", xlab = "size factors", cex = .6 )

par(mfrow=c(1,2))
## extracting normalized counts
counts.sf_normalized <- counts(DESeq.ds, normalized=TRUE)
## adding the boxplots
boxplot(counts(DESeq.ds), main = "read counts only", cex = .6)
boxplot(counts.sf_normalized, main = "SF normalized", cex = .6)


par(mfrow=c(1,2)) # to plot the two box plots next to each other
## bp of non-normalized
boxplot(log2(counts(DESeq.ds) +1), notch=TRUE,
main = "Non-normalized read counts",
ylab ="log2(read counts)", cex = .6)
## bp of size-factor normalized values
boxplot(log2(counts(DESeq.ds, normalized=TRUE) +1), notch=TRUE,
main = "Size-factor-normalized read counts",
ylab ="log2(read counts)", cex = .6)



## non-normalized read counts plus pseudocount
log.counts <- log2(counts(DESeq.ds, normalized = FALSE) + 1)
## instead of creating a new object, we could assign the values to a distinct matrix
## within the DESeq.ds object
assay(DESeq.ds, "log.counts") <- log2(counts(DESeq.ds, normalized = FALSE) + 1)
## normalized read counts
assay(DESeq.ds, "log.norm.counts") <- log2(counts(DESeq.ds, normalized=TRUE) + 1)
par(mfrow=c(1,2))
DESeq.ds[, c("Control_2","Control_3")] %>%
assay(., "log.norm.counts") %>%
plot(., cex=.1, main = "Control_2 vs Control_3")
DESeq.ds[, c("GPE_2","GPE_3")] %>%
assay(., "log.norm.counts") %>%
plot(., cex=.1, main = "GPE_2 vs GPE_3")

DESeq.rlog <- rlog(DESeq.ds, blind = FALSE) #since being treated with drug, I am trying blind
par(mfrow=c(1,2))
plot(assay(DESeq.ds, "log.norm.counts")[,1:2], cex=.1,
main = "size factor and log2-transformed")
## the rlog-transformed counts are stored in the "assay" accessor
plot(assay(DESeq.rlog)[,1:2],
cex=.1, main = "rlog transformed",
xlab = colnames(assay(DESeq.rlog[,1])),
ylab = colnames(assay(DESeq.rlog[,2])) )


save(file="./LNCap_Pou2f3_treatment/LNCap_DEseq.RData", DESeq.ds, DESeq.rlog)
```


## DEGs
```{r}
DESeq.ds %<>% DESeq()
```

## Load DEseq object
```{r}

library(DESeq2)
library(magrittr)

#load DEseq object
load("/Users/Caden/Desktop/angsd/project/angsd_project/DEseq/NPC_DEseq.RData")
DESeq.ds
```

## DE Analysis

### Run DE
```{r}
DESeq.ds %<>% DESeq()
```

### Check raw p value distrubtion 
```{r}
rowData(DESeq.ds)$WaldPvalue_Genotype_NRE_vs_CNE %>%
hist(breaks=19, main="Raw p-values for NRE vs CNE")

preOut_rawP <- rowData(DESeq.ds)$WaldPvalue_Genotype_NRE_vs_CNE
```
### Check logFC of significant genes
```{r}
DGE.results <- results(DESeq.ds,independentFiltering = TRUE, alpha = 0.05)
head(DGE.results)
summary(DGE.results)
table(DGE.results$padj < 0.05)

```

### Check adjusted p value distrubtiuon
```{r}
DGE.results$padj %>%
hist(breaks=19, main="Adjusted p-values for GPE vs Control")

preOut_adjP <- DGE.results$padj
```

### Sort DEGs
```{r}
DGE.results.sorted <- DGE.results %>% `[`(order(.$padj),)
DGE.results.sorted[seq(1:25),]
```

### Heatmap
```{r}
library(pheatmap)
# identify genes with the desired adjusted p-value cut-off
DGEgenes <- rownames(subset(DGE.results.sorted, padj < 0.05))
rlog.dge <- DESeq.rlog[DGEgenes,] %>% assay
pheatmap(rlog.dge, scale="row",
show_rownames=FALSE, main="DGE (row-based z-score)")
```

### Volcano Plot
```{r, fig.height=8}
library(EnhancedVolcano)
vp1 <- EnhancedVolcano(DGE.results,
lab=rownames(DGE.results),
x='log2FoldChange', y='padj',
pCutoff=0.05,
title="NRE / CNE")
print(vp1)
```
### Plot CHD6
```{r}
plotCounts(
  dds= DESeq.ds,
  gene="CHD6",
  intgroup = "Genotype"
  
)


plotCounts(
  dds= DESeq.ds,
  gene="CHD6",
  intgroup = "Treatment"
  
)

plotCounts(
  dds= DESeq.ds,
  gene="CHD6",
  intgroup = c("Genotype","Treatment")
  
)



plotCounts(
  dds= DESeq.ds,
  gene="IL10",
  intgroup = "Genotype"
  
)

plotCounts(
  dds= DESeq.ds,
  gene="SOX4",
  intgroup = "Genotype"
  
)


plotCounts(
  dds= DESeq.ds,
  gene="EHF",
  intgroup = "Genotype"
  
)


plotCounts(
  dds= DESeq.ds,
  gene="ASCL1",
  intgroup = "Genotype"
  
)


plotCounts(
  dds= DESeq.ds,
  gene="CHD7",
  intgroup = "Genotype"
  
)



plotCounts(
  dds= DESeq.ds,
  gene="E2F1",
  intgroup = "Genotype"
  
)


plotCounts(
  dds= DESeq.ds,
  gene="EZH2",
  intgroup = "Genotype"
  
)
```

```{r}
genes_interest <- c("CHD6", "EZH2", "E2F1")
DGE.results.sorted[rownames(DGE.results.sorted) %in% genes_interest,]
```

